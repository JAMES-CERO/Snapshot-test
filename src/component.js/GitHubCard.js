import { useEffect, useState } from 'react';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';

const GitHubCard = (props) => {

    const [githubName, setGithubName] = useState('')
    const [githubRepo, setGithubRepo] = useState('')
    const [githubAbout, setGithubAbout] = useState('')

    useEffect(() => {
        fetch('https://api.github.com/users/learningToCode1234')
        .then(res => res.json())
        .then(data => {
            console.log(data)
            setGithubName(data.name)
            setGithubRepo(data.repos_url)
            setGithubAbout(data.bio)
        })
    })
  return (
    <Card >
      <Card.Img variant="top" src="holder.js/100px180" />
      <Card.Body>
        <Card.Title> {githubName}</Card.Title>
        <Card.Text>
            {githubAbout}
        </Card.Text>
        <a href={githubRepo}><Button variant="primary" >Repositories</Button></a>
      </Card.Body>
    </Card>
  );
}

export default GitHubCard;

