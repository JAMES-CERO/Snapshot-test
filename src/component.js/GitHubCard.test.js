import GitHubCard from './GitHubCard'
import renderer from 'react-test-renderer'

test(' render a car scrrenshot', () => {
    const github = renderer.create(<GitHubCard/>).toJSON()
    expect(github).toMatchSnapshot()
})

