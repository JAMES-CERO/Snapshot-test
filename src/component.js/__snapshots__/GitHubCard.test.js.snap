// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` render a car scrrenshot 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="holder.js/100px180"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
       
    </div>
    <p
      className="card-text"
    />
    <a
      href=""
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="button"
      >
        Repositories
      </button>
    </a>
  </div>
</div>
`;
